x/bank/keeper/keeper.go

```diff
@@ -231,6 +239,14 @@ func (k BaseKeeper) UndelegateCoins(ctx sdk.Context, moduleAccAddr, delegatorAdd
 		return sdkerrors.Wrap(sdkerrors.ErrInvalidCoins, amt.String())
 	}
 
+	// if there are hooks, call the BeforeSend hook
+	if k.hooks != nil {
```

Comment:
Lets please add these hooks != nil check in the hooks struct itself, and not in this logic.

Reply:
Unless this is the standard for the rest of bank. If so, lets make a github issue for it to be fixed